Mathlib/Data/Seq/Seq.lean

```diff
@@ -716,6 +716,21 @@ def fold (s : Seq α) (init : β) (f : β → α → β) : Seq β :=
     | some (x, s) => .some (f acc x, f acc x, s)
   cons init <| corec f (init, s)
 
+/-- Applies `f` to the `n`th element of the sequence, if it exists, replacing that element
+with the result. -/
+def update (s : Seq α) (n : ℕ) (f : α → α) : Seq α where
+  val := Function.update s.val n ((s.val n).map f)
```

Comment:
Naïvely I would expect
```suggestion
def update (s : Seq α) (n : ℕ) (a : α) : Seq α where
  val := Function.update s.val n a
```
but maybe this doesn't make sense because streams can terminate?

Reply:
I see. Don't you think `update` should be defined in terms of `set`, then?

Reply:
Yeah sorry I thought about it again and decided my earlier suggestion was stupid